sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -508,164 +498,202 @@ export class EventHubSender extends LinkEntity {
     message: AmqpMessage | Buffer,
     options: SendOptions & EventHubProducerOptions = {}
   ): Promise<void> {
-
     const abortSignal: AbortSignalLike | undefined = options.abortSignal;
-    const sendEventPromise = () =>
+
+    const sendOperationPromise = () =>
       new Promise<void>((resolve, reject) => {
+        let waitTimer: any;
+
+        let onRejected: Func<EventContext, void>;
+        let onReleased: Func<EventContext, void>;
+        let onModified: Func<EventContext, void>;
+        let onAccepted: Func<EventContext, void>;
+        let onAborted: () => void;
+
         const rejectOnAbort = () => {
           const desc: string =
-            `[${this._context.connectionId}] The send operation on the Sender "${this.name}" with ` +
-            `address "${this.address}" has been cancelled by the user.`;
+            `[${this._context.connectionId}] The send operation on the Sender "${
+              this.name
+            }" with ` + `address "${this.address}" has been cancelled by the user.`;
           log.error(desc);
           reject(new AbortError("The send operation has been cancelled by the user."));
         };
 
-        if (abortSignal && abortSignal.aborted) {
```

Comment:
Why did you get rid of this abort check? This lets us exit early if the `abortSignal` already indicated a cancellation should occur.

Reply:
That's odd it should exist at line 581, per what I have on my local. 
Taking a look at commits